### ISSUE TITLE
SpikeGeneratorGroup

### ISSUE THREAD

**Issue by thesamovar** [brian-team/brian2#162]
I was going to write my first program in Brian 2 today, and realised I needed `SpikeGeneratorGroup`.

**Comment by mstimberg** [comment]
Just stumbled across this old issue -- at some point we should look into refactoring the generator expression parsing that we now have for synaptic connections and see whether we could re-use it in places like `SpikeGeneratorGroup`.

**Comment by thesamovar** [comment]
Good idea!

**Comment by thesamovar** [comment]
Not for 2.0 though, I think we'll agree.

**Comment by mstimberg** [comment]
> Not for 2.0 though, I think we'll agree.

Yes, fully agree. Apart from the refactoring work this issue also needs more discussion of the syntax, not something that we should rush.
